File: cpp/58d_1851B_ParitySort.cpp
#include<bits/stdc++.h>

using namespace std;

string solve(vector<int> arr, int n){
    vector<int> temp(arr.begin(), arr.end());
    sort(temp.begin(), temp.end());
    for(int i = 0; i < n; i++){
        if(temp[i] % 2 != arr[i] % 2){
            return "NO";
        }
    }
    return "YES";
}

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    int t, n, temp;
    vector<string> ans;
    cin>>t;

    for(int i = 0; i < t; i++){
        cin>>n;
        vector<int> arr;
        for (int j = 0; j < n; j++){
            cin>>temp;
            arr.push_back(temp);
        }
        ans.push_back(solve(arr, n));
    }

    for(auto i : ans){
        cout<<i<<'\n';
    }
    return 0;
}

File: python/58d_1851B_ParitySort.py
def solve(arr, n):
    temp = list(arr)
    temp.sort()
    for i in range(n):
        if arr[i] % 2 != temp[i] % 2:
            return "NO"
    return "YES"

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        arr = list(map(int, input().split()))
        ans.append(solve(arr, n))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()